### PR TITLE
Adds helper text to Output path in create form

### LINKED ISSUE #94: Improve placeholder text for Output prefix in job creation form
### Problem

The create job API takes an `output_prefix` argument that determines in which directory the output files will be put when the job completes. Currently the `TextField` placeholder for this UI element reads "Output prefix". We should improve that to give the user more useful information with example. This information should include:

* What the output prefix is (where output files will go)
* What its format is (relative or absolute path)
* Default value
* Examples.

### PR DESCRIPTION
**Pull request by jweill-aws**
Fixes #94. Adds helper text to "Output path" input.

image

### CODE REVIEW

**Comment by ellisonbg** [comment]
Question for @3coins on the backend. Is the output prefix:

a. The path of output files relative to the input file?
b. The path relative to the root of the Jupyter Server?
c. The absolute path on the file system?
d. A combination of any of the above?

File: src/mainviews/create-job.tsx [hunk 1]
```diff
@@ -680,6 +680,9 @@ export function CreateJob(props: ICreateJobProps): JSX.Element {
             value={props.model.outputPath}
             id={`${formPrefix}outputPath`}
             name="outputPath"
+            helperText={trans.__(
+              'Relative path where output files are written'
```
Comment (3coins) [inline src/mainviews/create-job.tsx:684]:
```suggestion
              'Output directory where job output files are written, default is same directory as input file'
```

**Comment by 3coins** [comment]
> Question for @3coins on the backend. Is the output prefix:
> 
> a. The path of output files relative to the input file? b. The path relative to the root of the Jupyter Server? c. The absolute path on the file system? d. A combination of any of the above?

Answer is b) The path relative to the root of the Jupyter Server

**Comment by ellisonbg** [comment]
Do we accept paths with a leading `.` or `..`? If the path is always relative to the server root, we should show the path relative to the input notebook.

**Comment by dlqqq** [comment]
From 10/17/2022 standup:

1. Let's consider renaming "output path" to "output directory" in the UI while we're at it. Paths can refer both to directories and regular files, so let's make our intentions explicit for clarity.

2. We should add a "home" icon with a forward slash to the left of the output directory input to further emphasize that the path must be relative to the server root. @ellisonbg Please confirm.

File: src/mainviews/create-job.tsx (hunk 1 above)
Comment (dlqqq) [inline src/mainviews/create-job.tsx:684]: Believe a more helpful helper text would be: "Path to directory in which output files will be written. Path must be relative to the server root." @ellisonbg Thoughts?

**Review by dlqqq — CHANGES_REQUESTED** [review]

**Comment by jweill-aws** [comment]
Per @ellisonbg's feedback, sets the default output directory to the input file's directory. Relies on the invariant in `path` that directories are delimited with `/`.

File: src/index.tsx [hunk 2]
```diff
@@ -87,20 +87,34 @@ function getSelectedItem(widget: FileBrowser | null): Contents.IModel | null {
   return firstItem;
 }
 
-function getSelectedFilePath(widget: FileBrowser | null): string | null {
+// Get only the file name, with no parent directories, of the currently selected file.
+function getSelectedFileName(widget: FileBrowser | null): string | null {
```
Comment (jweill-aws) [inline src/index.tsx:91]: These now appear in the same order in which they're called below.

**Comment by ellisonbg** [comment]
Can you post a screenshot of the current state of this PR? Here are the main points I want to make sure we are covering:

* The Input file should show the path of the input file, not just its name.ext.
* The default value of the output directory (let's call it that in the UI) should be the path of the input file.
* To emphasize to users that these paths are relative to the server root, we should render a `:home: / ` to the left of the input file and output directory as the home icon is used in JupyterLab to indicate the server root directory.

**Review by dlqqq — CHANGES_REQUESTED** [review]
This looks great! Really close to what we want. Just two more things:

1. We would like to have a home icon followed by a forwards slash to the left of all input fields that accept relative paths to the server root, i.e. "input file" and "output directory".

2. Could you add a forwards slash to the output directory input value to further emphasize that it's a directory? For example, in the below screenshot, it should say "outputs/". The backend should be able to handle this redundant forwards slash, but please test it out as well when you make this change.

Screen Shot 2022-10-20 at 10 52 24 AM

**Comment by jweill-aws** [comment]
> To emphasize to users that these paths are relative to the server root, we should render a `:home: / ` to the left of the input file and output directory as the home icon is used in JupyterLab to indicate the server root directory.

@ellisonbg Should the ":house:/" be rendered to the left of the outlined "Output directory" `TextField` or should it be _inside_ the input area?

**Comment by ellisonbg** [comment]
Nice! I think putting the home icon inside the text field is nice. Also, let's add it to the input file text field as well.

**Comment by ellisonbg** [comment]
Also, let's make sure to render these things in a similar way in the detail view.

**Comment by jweill-aws** [comment]
Here's what the UI looks like with the :house: adornment in both text fields

image

**Comment by jweill-aws** [comment]
… and in the detail view:

image

**Pull request merged**